tests/unit/test_kml_conversion.py

```diff
@@ -0,0 +1,265 @@
+    )
+    assert result
+    assert new_york_city is not None
+
+    yield [
+        san_francisco,
+        los_angeles,
+        new_york_city,
+    ]
+
+
+@pytest.fixture
+def positions() -> list[position_global_relative_altitude.PositionGlobalRelativeAltitude]:  # type: ignore
+    """
+    List of locations.
+    """
+    result, san_francisco = position_global_relative_altitude.PositionGlobalRelativeAltitude.create(
+        37.7749, -122.4194, 0.0
+    )
+    assert result
+    assert san_francisco is not None
+
+    result, los_angeles = position_global_relative_altitude.PositionGlobalRelativeAltitude.create(
+        34.0522, -118.2437, 0.0
+    )
+    assert result
+    assert los_angeles is not None
+
+    result, new_york_city = position_global_relative_altitude.PositionGlobalRelativeAltitude.create(
+        40.7128, -74.0060, 0.0
+    )
+    assert result
+    assert new_york_city is not None
+
+    yield [
+        san_francisco,
+        los_angeles,
+        new_york_city,
+    ]
+
+
+@pytest.fixture
+def named_positions() -> list[position_global_relative_altitude.NamedPositionGlobalRelativeAltitude]:  # type: ignore
+    """
+    List of named locations.
+    """
+    result, san_francisco = (
+        position_global_relative_altitude.NamedPositionGlobalRelativeAltitude.create(
+            "San Francisco", 37.7749, -122.4194, 0.0
+        )
+    )
+    assert result
+    assert san_francisco is not None
+
+    result, los_angeles = (
+        position_global_relative_altitude.NamedPositionGlobalRelativeAltitude.create(
+            "Los Angeles", 34.0522, -118.2437, 0.0
+        )
+    )
+    assert result
+    assert los_angeles is not None
+
+    result, new_york_city = (
+        position_global_relative_altitude.NamedPositionGlobalRelativeAltitude.create(
+            "New York City", 40.7128, -74.0060, 0.0
+        )
+    )
+    assert result
+    assert new_york_city is not None
+
+    yield [
+        san_francisco,
+        los_angeles,
+        new_york_city,
+    ]
+
+
+def test_named_locations_to_kml_with_save_path(
+    named_locations: list[location_global.NamedLocationGlobal], tmp_path: pathlib.Path
+) -> None:
+    """
+    Basic test case to save KML to the correct path when provided.
+    """
+    # Setup
+    expected_kml_document_path = pathlib.Path(PARENT_DIRECTORY, "expected_named_locations.kml")
+    actual_kml_document_name = "actual"
+
+    tmp_path.mkdir(parents=True, exist_ok=True)
+
+    # Run
+    result, actual_kml_file_path = kml_conversion.named_locations_to_kml(
+        named_locations,
+        actual_kml_document_name,
+        tmp_path,
+    )
+
+    # Check
+    assert result
+    assert actual_kml_file_path is not None
+
+    assert actual_kml_file_path.exists()
+    assert actual_kml_file_path.suffix == KML_SUFFIX
+
+    assert actual_kml_file_path.read_text(encoding="utf-8") == expected_kml_document_path.read_text(
+        encoding="utf-8"
+    )
+
+
+def test_locations_to_kml(
```

Comment:
Just curious why this one doesn't have "with_save_path" in its name

Reply:
Refactored (added mock).